sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -694,14 +699,16 @@ def read(self, range_tracker):
     raise NotImplementedError('BigQuery source must be split before being read')
 
   def _setup_temporary_dataset(self, bq):
+    query = self.evaluate_query_valueprovider()
```

Comment:
There's a `check_accessible` decorator that could be used for this function (as well as for `estimate_size`).
I think you could also test if `self.query` is a ValueProvider or not in the constructor. If it's not, you can then create a StaticValueProvider object. `self.evaluate_query_valueprovider()` would be unnecessary.

Here'a en example:
https://github.com/apache/beam/blob/f75838f306fc1d851b0dea11c2cb861b43430568/sdks/python/apache_beam/io/filebasedsource.py#L115-L117

Reply:
Thanks for the tip re: `check_accessible`, I didn't know about that. I've updated the PR based on these suggestions.